#! /usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import *

PACKAGE = "param_pkg"

# 创建一个参数生成器
gen = ParameterGenerator()

# 添加参数说明，便于后续生成界面
#        参数名           类型           等级       参数描述          默认值              最小值     最大值
# 仿真参数
# gen.add("KD1",          double_t,	    0,		   "KD1",		    0,			      0,		 5)
# gen.add("KD2",          double_t,	    0,		   "KD2",		    0,			      0,		 5)
# gen.add("KD3",          double_t,	    0,		   "KD3",		    0.01,			  0,		 5)

# gen.add("lamda",	    double_t,		0,		   "lamda",		    25.0,			  0,		 50)

# gen.add("Gamma",	    double_t,		0,	       "Gamma",		    2.0,			  0,		 30)

# gen.add("KDTu1",	    double_t,		0,		   "KDTu",		    1.5,			  0,		 5)
# gen.add("KDTu2",	    double_t,		0,		   "KDTu",		    1.5,			  0,		 5)
# gen.add("KDTu3",	    double_t,		0,		   "KDTu",		    0.5,			  0,		 5)

# gen.add("lamdaTu1",	    double_t,		0,	       "lamdaTu",		25.0,			  0,		 50)
# gen.add("lamdaTu2",	    double_t,		0,	       "lamdaTu",		25.0,			  0,		 50)
# gen.add("lamdaTu3",	    double_t,		0,	       "lamdaTu",		5.0,			  0,		 50)

# gen.add("GammaTu1",	    double_t,		0,		   "GammaTu",		1.5,			  0,		 20)
# gen.add("GammaTu2",	    double_t,		0,		   "GammaTu",		1.5,			  0,		 20)
# gen.add("GammaTu3",	    double_t,		0,		   "GammaTu",		0.7,			  0,		 20)
# 实机参数
gen.add("KD1",          double_t,	    0,		   "KD1",		    0,			      0,		 5)
gen.add("KD2",          double_t,	    0,		   "KD2",		    0,			      0,		 5)
gen.add("KD3",          double_t,	    0,		   "KD3",		    0.02,			  0,		 5)

gen.add("lamda",	    double_t,		0,		   "lamda",		    7.0,			  0,		 50)

gen.add("Gamma",	    double_t,		0,	       "Gamma",		    10.0,			  0,		 30)

gen.add("KDTu1",	    double_t,		0,		   "KDTu",		    1.5,			  0,		 5)
gen.add("KDTu2",	    double_t,		0,		   "KDTu",		    1.5,			  0,		 5)
gen.add("KDTu3",	    double_t,		0,		   "KDTu",		    0.5,			  0,		 5)

gen.add("lamdaTu1",	    double_t,		0,	       "lamdaTu",		6,			  0,		 50)
gen.add("lamdaTu2",	    double_t,		0,	       "lamdaTu",		8,			  0,		 50)
gen.add("lamdaTu3",	    double_t,		0,	       "lamdaTu",		2,			  0,		 50)

gen.add("GammaTu1",	    double_t,		0,		   "GammaTu",		1.5,			  0,		 20)
gen.add("GammaTu2",	    double_t,		0,		   "GammaTu",		1.5,			  0,		 20)
gen.add("GammaTu3",	    double_t,		0,		   "GammaTu",		0.7,			  0,		 20)

# 调用生成器生成config配置文件
#                   包名   节点名称    生成文件名
exit(gen.generate(PACKAGE, PACKAGE, "param"))